### DISCUSSION TITLE
Old Data returned in JSON for a call to  https://api.weather.gov/gridpoints/RAH/43,69/forecast

### DISCUSSION THREAD

**Discussion by gmstuff** [weather-gov/api#286] [Q&A]
Hi - I hope its me just doing something wrong. But...
I have some   python code that pulls data from api.weather.gov that the JSON returned is 3 days old.

My question is : Why is data from 3 days ago being returned in the JSON? Or am I doing something wrong? Please :-)

My code pulls from this URL https://api.weather.gov/gridpoints/RAH/43,69/forecast
and then parses the JSON.
However the data appears to contain data from 3 days ago. 

Below is the code pulled out of the larger script. The results are printed to the screen in the Try/Except block. The results (shown  below for the UPDATED value is 09/21/2020  whereas the date today is 09/24/2020. I have noticed this for the past few days and tried multiple debug bits of code to zero in on the issue to no avail

**Screen shot
![Returned_JSON_Data_Old](https://user-images.githubusercontent.com/71852099/94204154-bb55e080-fe8e-11ea-89e5-47ea7175ad39.JPG)
[Returned_JSON_With_OldData.txt](https://github.com/weather-gov/api/files/5279400/Returned_JSON_With_OldData.txt)

** attached showing the returned JSON the system date/time and the returned data if using a browser from the same machine.
Attached is also the output of the JSON printed to screen

The code snippet below could be used to reproduce
I was expecting the data returned in the JSON to be pretty much the same as if I navigate using a browser to the same URL

**Code snippet** 
    	URLHeaders = {
	   "Cache-Control":"no-cache",
	   "Pragma": "no-cache"
       	}
    #URLHEaders tried adding URLHeaders as I am getting an old version of the JSON data from weather.gov in the initial request

        url = "https://api.weather.gov/gridpoints/RAH/43,69/forecast"
        print("URL to pull <" + url + ">")
        try:
            response = requests.get(url, headers=Win.URLHeaders)
            GetRawWeatherDataStatus = "Pass"
  
            gdt, gtm, ldt, ltm = self.GetDateTime()
            gdt = gdt[0:4] + '/' + gdt[4:6] + '/' + gdt[6:8]
            gtm = gtm[0:2] + ':' + gtm[2:4] + ':' + gtm[4:6] + ' Z'  
            ldt = ldt[0:4] + '/' + ldt[4:6] + '/' + ldt[6:8]
            ltm = ltm[0:2] + ':' + ltm[2:4] + ':' + ltm[4:6]
            print("URL Pulled @ " + str(ldt) + " " + str(ltm))
            print (response)
            
                                    
            
        except Exception as ex:
            print("Exception <" + str(ex) + ">")
            if (str(ex).find("NewConnectionError") != -1):
                print ("response = requests.get(url, headers=Win.URLHeaders)  " + url)
                GetRawWeatherDataStatus ="Fail"
                return
***** end of code snippet

Printed returned JSON snippet showing hte data is updated 09/21/2020 - which is 3 days ago.

{'@context': ['https://geojson.org/geojson-ld/geojson-context.jsonld', {'@version': '1.1', 'wx': 'https://api.weather.gov/ontology#', 'geo': 'http://www.opengis.net/ont/geosparql#', 'unit': 'http://codes.wmo.int/common/unit/', '@vocab': 'https://api.weather.gov/ontology#'}], 'type': 'Feature', 'geometry': {'type': 'Polygon', 'coordinates': [[[-79.448826, 36.153154], [-79.4519998, 36.1309056], [-79.42445029999999, 36.1283402], [-79.42127099999999, 36.150588299999995], [-79.448826, 36.153154]]]}, 'properties': {'updated': '2020-09-21T19:42:18+00:00',

Environment:
This is running on a Raspberry PI 4b with Version 10 Buster OS
using Python 3

**Comment by JJKraw** [comment]
See https://github.com/weather-gov/api/issues/71

**Reply by joshakaw** [reply]
For anyone reading this after they switched from Issues to Discussions, here's the link:
https://github.com/weather-gov/api/discussions/329

**Comment by scadergit** [comment]
Thanks for the report. Closing as duplicate of #71.
